### ISSUE TITLE
Support for auxillary/sidechain dsp_node I/O

### ISSUE THREAD

**Issue by skrytt** [skrytt/baselisk-rs#1]
The broad goal here is to support using aux/sidechain input and output buffers for DSP nodes.

Conceptually this is a little closer to how analog synthesizers work. Consider things like oscillator pitch and sync signals. A DSP unit applies a function to all of these inputs to generate its outputs.

The current design overuses methods to modify state of DSP units. A particular problem with the current implementation is that the methods can only be triggered at the start of each block. Use of buffers for such events would allow sample-based events to occur and, hopefully, means that the block size has less (or ideally no) impact on the output signal.

**Comment by skrytt** [comment]
As I've explored the design of my solution, I've come to the conclusion that my original idea here wasn't the best possible approach. However, I found an alternative that seems to work quite well.

The alternative involves using an iterator of EngineEvents (things like note changes, pitch bends, etc) to determine keyframes in processing and change the state as required at those points; then to iterate between keyframes to generate samples.
